pypesto/problem.py

```diff
@@ -149,15 +148,117 @@ def normalize_input(self):
             raise AssertionError("lb dimension invalid.")
         if self.ub.size != self.dim:
             raise AssertionError("ub dimension invalid.")
-        if self.x_guesses.shape[1] != self.dim:
-            raise AssertionError("x_guesses form invalid.")
+        if check_x_guesses:
+            if self.x_guesses.shape[1] != self.dim:
+                raise AssertionError("x_guesses form invalid.")
         if len(self.x_names) != self.dim_full:
             raise AssertionError("x_names must be of length dim_full.")
         if len(self.x_fixed_indices) != len(self.x_fixed_vals):
             raise AssertionError(
                 "x_fixed_indices and x_fixed_vals musti have the same length."
             )
 
+    def fix_parameters(self, parameter_indices, parameter_vals):
+        """
+        Fix specified parameters to specified values
+        """
+        if not isinstance(parameter_indices, list):
+            parameter_indices = [parameter_indices]
+
+        if not parameter_vals is list:
+            parameter_vals = [parameter_vals]
+
+        lb = []
+        ub = []
+
+        # first clean to be fixed indices to avoid redundancies
+        for i_index, i_parameter in enumerate(parameter_indices):
+            # check if parameter was already fixed, otherwise add it to the
+            # fixed parameters
+            if i_parameter in self.x_fixed_indices:
+                self.x_fixed_vals[self.x_fixed_indices.index(i_parameter)] = \
+                    parameter_vals.pop(i_index)
```

Comment:
hm. does this work when there are multiple cases where i_parameter in self.x_fixed_indices? when you pop a value, it means that parameter_vals length decreases, such that i_index does not point to the correct element between parameter_indices and parameter_vals?